### ISSUE TITLE
shuffle and batch size option in test process affect classification probability and accuracy

### ISSUE THREAD

**Issue by sainisanjay** [facebook/C3D#277]
I found something weird about testing pre-trained c3d-resnet model(trained on ucf101) while changing the shuffle and batch size option in the test process.

when I tested the model with shuffle option as false, the probability scores (which are used for calculating top-k accuracy) from accuracy layer showed to be almost 0, regardless of test batch size. (ex) 0.05xx) 

On the other hand, when I tested with the shuffle option as true, the probability scores from the accuracy layer showed to be reasonable (over 0.9 for highest classification score with correct prediction). 
Also, the probability score values were proportional to batch size (test with more batch size resulted in higher value of probability scores). 

However, If I use batch size 1 for test, regardless of shuffle option, the probability scores were very low and the most of cases were classified as class 57 (regardless of the class labels).

Is there anyone who had similar results like this? I want to know why and how the shuffle and batch size option affect the test process of C3D.

**Comment by dutran** [comment]
Which prototxt file did you use for testing? Did you set, use_global_stats: true? If not, batch norm will not do anything with batch size =1 or not doing right if not shuffle.

**Comment by sainisanjay** [comment]
I used the train_resnet18_r2.prototxt and solver_r2.prototxt in the examples/c3d_ucf101_finetuning folder. I just changed the shuffle and batch size option, and used default setting for all other layer settings. (I used use_global_status: false)
can you explain more about the relation between batch norm and shuffle?
